datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -627,15 +629,34 @@ impl<'a> ConstEvaluator<'a> {
             return ConstSimplifyResult::NotSimplified(s);
         }
 
+        let start_type = match expr.get_type(&self.input_schema) {
+            Ok(t) => t,
+            Err(err) => return ConstSimplifyResult::SimplifyRuntimeError(err, expr),
+        };
+
         let phys_expr =
             match create_physical_expr(&expr, &self.input_schema, self.execution_props) {
                 Ok(e) => e,
                 Err(err) => return ConstSimplifyResult::SimplifyRuntimeError(err, expr),
             };
+
         let col_val = match phys_expr.evaluate(&self.input_batch) {
             Ok(v) => v,
             Err(err) => return ConstSimplifyResult::SimplifyRuntimeError(err, expr),
         };
+
+        // TODO(@notfilippo): a fix for the select_arrow_cast error
+        let end_type = col_val.data_type();
+        if end_type.logically_eq(&start_type) && start_type != end_type {
+            return ConstSimplifyResult::SimplifyRuntimeError(
+                DataFusionError::Execution(format!(
```

Comment:
we could use `exec_err`

Reply:
`exec_err` returns a result but I could use `exec_datafusion_err`. thx!